test/axios2.test.js

```diff
@@ -595,4 +596,393 @@ describe("Axios AuthHttpRequest class tests", function() {
+                assertEqual(accessTokenPayload._jwtPName, "jwt");
+                assertEqual(accessTokenPayload.iss, "http://0.0.0.0:8080");
+                assertEqual(accessTokenPayload.customClaim, "customValue");
+
+                let jwt = accessTokenPayload.jwt;
+
+                // Decode the JWT
+                let decodeResponse = await axios.post(`${BASE_URL}/jsondecode`, JSON.stringify({ jwt }), {
+                    headers: {
+                        Accept: "application/json",
+                        "Content-Type": "application/json"
+                    }
+                });
+
+                let decodedJWT = await decodeResponse.data;
+
+                // Verify the JWT claims
+                assertEqual(decodedJWT.sub, userId);
+                assertEqual(decodedJWT._jwtPName, undefined);
+                assertEqual(decodedJWT.iss, "http://0.0.0.0:8080");
+                assertEqual(decodedJWT.customClaim, "customValue");
+
+                await axios.post(
+                    `${BASE_URL}/resetST`,
+                    JSON.stringify({
+                        jwtPropertyName: "customJWTProperty"
+                    }),
+                    {
+                        headers: {
+                            Accept: "application/json",
+                            "Content-Type": "application/json"
+                        }
+                    }
+                );
+
+                let attemptRefresh = await supertokens.attemptRefreshingSession();
+                assertEqual(attemptRefresh, true);
+
+                // Get access token payload
+                accessTokenPayload = await supertokens.getAccessTokenPayloadSecurely();
+
+                // Verify new access token payload
+                assertEqual(accessTokenPayload.jwt, undefined);
+                assertNotEqual(accessTokenPayload.customJWTProperty, undefined);
+                assertEqual(accessTokenPayload.sub, userId);
+                assertEqual(accessTokenPayload._jwtPName, "customJWTProperty");
+                assertEqual(accessTokenPayload.iss, "http://0.0.0.0:8080");
+                assertEqual(accessTokenPayload.customClaim, "customValue");
+
+                jwt = accessTokenPayload.customJWTProperty;
+
+                decodeResponse = await axios.post(`${BASE_URL}/jsondecode`, JSON.stringify({ jwt }), {
+                    headers: {
+                        Accept: "application/json",
+                        "Content-Type": "application/json"
+                    }
+                });
+
+                decodedJWT = await decodeResponse.data;
+
+                // Verify new JWT
+                assertEqual(decodedJWT.sub, userId);
+                assertEqual(decodedJWT._jwtPName, undefined);
+                assertEqual(decodedJWT.iss, "http://0.0.0.0:8080");
+                assertEqual(decodedJWT.customClaim, "customValue");
+            });
+        } finally {
+            await browser.close();
+        }
```

Comment:
Add a new test that will:
- Create a new session with a JWT
- Wait for the access token to expire
- Call getAccessTokenPayloadSecurely and get the JWT from it
- Make sure that the fetched JWT is a refreshed one and that the refresh API was called exactly once.

Reply:
Done, added for both fetch and axios